ambry-api/src/main/java/com/github/ambry/account/Dataset.java

```diff
@@ -0,0 +1,175 @@
+/*
+ * Copyright 2023 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ */
+package com.github.ambry.account;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+import java.util.Collections;
+import java.util.Map;
+
+
+/**
+ * <p>
+ *  A dataset represents a logical data entity (e.g., a model, or an index, or some collection of directories). Like
+ * accounts, a dataset needs to be on-boarded before any data can be stored in it, but unlike accounts which are on-boarded
+ * via Nuage UI, datasets can be on-boarded by the account owner via rest request. A dataset has some metadata associated
+ * with it, that defines either some characteristics of the dataset (e.g., what kind of versioning scheme would be used
+ * to version that dataset) or some specification to Ambry as to how this dataset is to be handled by Ambry (e.g., delete
+ * any version of this dataset that is more than N days old).
+ * </p>
+ * <p>
+ *   Dataset name is provided by a user as an external reference to the dataset. Dataset name has to be distinct within
+ *   the same {@link Account}, but can be the same across different {@link Account}s.
+ * </p>
+ * <pre><code>
+ *  {
+ *    "accountName": "MyPrivateAccount",
+ *    "containerName": "MyPrivateContainer",
+ *    "datasetName": "MyPrivateDataset",
+ *    "versionSchema": "TIMESTAMP",
+ *    "expirationTimeMs": -1,
+ *    "retentionCount": 10,
+ *    "userTags": "{userTag1:tagValue1}"
+ *  } * </code></pre>
+ */
+@JsonDeserialize(builder = DatasetBuilder.class)
+public class Dataset {
+  //constant
+  static final String ACCOUNT_NAME_KEY = "accountName";
+  @JsonProperty(ACCOUNT_NAME_KEY)
```

Comment:
Are the annotations on the wrong line?  It's supposed to be above the declared field.

Reply:
It seems like this is not affect my serialization and deserialization. But thanks for point it out, I've updated it.